build/mk_version_hrl

```diff
@@ -1,36 +1,40 @@
-#! /usr/bin/env bash
+#!/usr/bin/env escript
+%% -*- erlang -*-
 
-if ! git rev-parse --is-inside-work-tree >/dev/null 2>&1
-then
-    echo "ERROR: Building outside a git work tree is not supported," >&2
-    echo "       unless the source has first been exported." >&2
-    echo "       This is for the version number to get correct." >&2
-    echo "       To build outside a git work tree, first export it:" >&2
-    echo "       Start with a git clone, then run helpers/export-from-git." >&2
-    echo "       For further info, see the README.md, the section section" >&2
-    echo "       named Building outside of a git work tree" >&2
-    echo "" >&2
-    echo "       Github automatically generates tar/zip files from tags,"  >&2
-    echo "       but these will unfortunately not do, since the version"  >&2
-    echo "       number does not get set correctly."  >&2
-    exit 1
-fi
 
-vsn="$(git describe --always --tags --match '[0-9]*.[0-9]*')"
 
-if [[ x"$vsn" = x || x"$vsn" =~ x[a-f0-9]{7,} ]]
-then
-    echo "ERROR: Unexpected version \"$vsn\", please make sure you are" >&2
-    echo "       building in a git repository with all tags intact, or" >&2
-    echo "       in a directory that has been exported from git." >&2
-    echo "       For further info, see the README.md, the section section" >&2
-    echo "       named Building outside of a git work tree" >&2
-    exit 1
-fi
+-mode(compile).
+-define(NOT_IN_GIT_ERR_MSG,
+"ERROR: Building outside a git work tree is not supported,
+       unless the source has first been exported.
+       This is for the version number to get correct.
+       To build outside a git work tree, first export it:
+       Start with a git clone, then run helpers/export-from-git.
+       For further info, see the README.md, the section section
+       named Building outside of a git work tree
+       Github automatically generates tar/zip files from tags,
+       but these will unfortunately not do, since the version
+       number does not get set correctly.").
 
-sed \
-    -e "s/@vsn@/$vsn/g" \
-    -e "s/is expected to be/was/g" \
-    -e '/^%% The version below/ i \
-%% DO NOT EDIT -- generated from gpb_version.hrl.in' \
-    -e "/^%% NB: The build.mk_version_hrl depends/ d"
+
+
+main(_) ->
+    InGit = os:cmd("git rev-parse --is-inside-work-tree  2>&1"),
+    case InGit of
+        "true\n" ->
+            next;
+        _ ->
+            erlang:throw(?NOT_IN_GIT_ERR_MSG)
```

Comment:
When it happens, this message gets printed as one long line with embedded `\n`. Maybe a `io:format(standard_error, "~s", [?NOT_IN_GIT_ERR_MSG])` preceding this line plus a shorter `throw` term, maybe just an atom?